File: TouchControl/src/MousePad.java
import java.awt.*;

import org.opencv.core.*;
import org.opencv.core.Point;

public class MousePad extends Pad
{
	private Robot mouseMover;

	private final double SCREEN_WIDTH = Toolkit.getDefaultToolkit().getScreenSize().getWidth();
	private final double SCREEN_HEIGHT = Toolkit.getDefaultToolkit().getScreenSize().getHeight();

	//Constructors
	public MousePad(Rect dimensions)
	{
		this(dimensions, DEFAULT_COLOR);
	}

	public MousePad(Rect dimensions, Scalar color)
	{
		super(dimensions, color);
		init();
	}

	private void init()
	{
		createRobot();
	}

	//Setters

	//Getters

	//Class methods
	@Override
	public void performAction()
	{
		if (hasDetection())
		{
			//Convert point in range to point on screen
			Point adjustedMousePoint = adjustPoint();

			//Move mouse pointer to point
			moveMouse(adjustedMousePoint);
		}
	}

	private Point adjustPoint()
	{
		//Local variables
		Point adjustedPoint = new Point();
		double widthAdjustment = SCREEN_WIDTH / dimensions.width;
		double heightAdjustment = SCREEN_HEIGHT / dimensions.height;

		//Adjust point
		adjustedPoint.x = detectionPoint.x * widthAdjustment;
		adjustedPoint.y = detectionPoint.y * heightAdjustment;

		return adjustedPoint;
	}

	private void moveMouse(Point point)
	{
		//Use robot to move mouse
		mouseMover.mouseMove((int) point.x, (int) point.y);
	}

	private void createRobot()
	{
		try
		{
			mouseMover = new Robot();
		}
		catch (AWTException e)
		{
			e.printStackTrace();
		}
	}

	@Override
	public String toString()
	{
		return super.toString() + String.format(TO_STRING_FORMAT, "Screen height:", this.SCREEN_HEIGHT)
				+ String.format(TO_STRING_FORMAT, "Screen width:", this.SCREEN_WIDTH);
	}
}

File: src/main/kotlin/me/jakejmattson/touchcontrol/touchables/MousePad.kt
package me.jakejmattson.touchcontrol.touchables

import me.jakejmattson.touchcontrol.utils.*
import org.opencv.core.*
import org.opencv.core.Point
import java.awt.*

/**
 * Demo Pad - Example application of abstract pad.
 * Controls mouse pointer position.
 *
 * @author JakeJMattson
 */
class MousePad(dimensions: Rect, color: Scalar) : Pad(dimensions, color) {
    private var mouseMover: Robot = Robot()

    override fun performAction() {
        if (hasDetection()) {
            //Convert point in range to point on screen
            val mousePoint = adjustPoint()

            //Move mouse pointer to point
            mouseMover.mouseMove(mousePoint)
        }
    }

    /**
     * Convert the internal detection point from a point on the Touchable to a
     * point on the screen so the mouse can be moved anywhere on the display.
     *
     * @return Adjusted point
     */
    private fun adjustPoint(): Point {
        val ratioPoint = Point(SCREEN_WIDTH / dimensions.width, SCREEN_HEIGHT / dimensions.height)

        return detectionPoint!! * ratioPoint
    }

    override fun toString() = super.toString() +
        format("Screen height:", SCREEN_HEIGHT) +
        format("Screen width:", SCREEN_WIDTH)

    companion object {
        private val SCREEN_WIDTH = Toolkit.getDefaultToolkit().screenSize.getWidth()
        private val SCREEN_HEIGHT = Toolkit.getDefaultToolkit().screenSize.getHeight()
    }
}